app/src/main/kotlin/co/netguru/baby/monitor/client/feature/client/home/ClientHomeActivity.kt

```diff
@@ -32,69 +38,98 @@ class ClientHomeActivity : DaggerAppCompatActivity(),
+        ViewModelProviders.of(this, factory)[ConfigurationViewModel::class.java]
+    }
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_client_home)
 
         setupView()
-        getData()
-        initWebSocketConnection()
-        checkInternetConnection()
-    }
+        setupObservers()
 
-    private fun checkInternetConnection() {
+        homeViewModel.fetchLogData()
         homeViewModel.checkInternetConnection()
-        homeViewModel.internetConnectionAvailability.observe(this, Observer { isConnected ->
-            if (!isConnected) showNoInternetSnackbar()
-        })
     }
 
-    override fun onSupportNavigateUp() =
-        findNavController(R.id.clientDashboardNavigationHostFragment).navigateUp()
-
-    private fun initWebSocketConnection() {
-        homeViewModel.selectedChild.observeNonNull(this, { child ->
+    private fun setupObservers() {
+        homeViewModel.selectedChild.observeNonNull(this) { child ->
             Timber.i("Opening socket to ${child.address}.")
             homeViewModel.openSocketConnection(URI.create(child.address))
-        })
-    }
-
-    private fun setupView() {
-        toolbarSettingsIbtn.setOnClickListener {
-            homeViewModel.shouldDrawerBeOpen.postValue(true)
         }
-        toolbarBackBtn.setOnClickListener {
-            findNavController(R.id.clientDashboardNavigationHostFragment).navigateUp()
-        }
-    }
 
-    private fun getData() {
-        homeViewModel.fetchLogData()
-        homeViewModel.selectedChild.observeNonNull(this, { child ->
-            setSelectedChildName(child.name ?: "")
-            GlideApp.with(this)
-                .load(child.image)
-                .placeholder(R.drawable.child)
-                .apply(RequestOptions.circleCropTransform())
-                .into(toolbarChildMiniatureIv)
+        homeViewModel.internetConnectionAvailability.observe(this, Observer { isConnected ->
+            if (!isConnected) showNoInternetSnackbar()
         })
+
+        homeViewModel.selectedChild.observeNonNull(this) { child ->
+            handleSelectedChild(child)
+        }
         homeViewModel.toolbarState.observe(this, Observer(this::handleToolbarStateChange))
+
         homeViewModel.backButtonState.observe(
             this,
             Observer {
                 backIbtn.setVisible(it?.shouldBeVisible == true)
                 setBackButtonClick(it?.shouldShowSnoozeDialog == true)
             })
+
         homeViewModel.shouldDrawerBeOpen.observe(this, Observer { shouldClose ->
-            if (shouldClose == true) {
-                client_drawer.openDrawer(GravityCompat.END)
-            } else {
-                client_drawer.closeDrawer(GravityCompat.END)
+            handleDrawerEvent(shouldClose)
+        })
+
+        homeViewModel.webSocketAction.observe(this, Observer {
+            when (it) {
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/netguru/baby-monitor-client-android/blob/b1aa35793dca6d7714e5534d6167592f0bbe6a99/app/src/main/kotlin/co/netguru/baby/monitor/client/feature/client/home/ClientHomeActivity.kt#L83"></span>Prefer using ‘if’ for binary conditions instead of ‘when’.</td>
    </tr>
  </tbody>
</table>